doc/dev/kdb-record.md

```diff
@@ -0,0 +1,110 @@
+# `kdb record` Command
+
+The `kdb record` tooling enables the user to record configuration sessions. It also allows exporting and importing the configuration changes.
+
+Recording sessions are stored on a per-user basis. This implies that a recording session can only track changes made by a single user.
+
+## Session handling
+
+* `kdb record start [session_name]`: Starts a new recording session. If no session name is provided, a new random name is generated. If a session with the provided name already exists, changes will be appended.
+* `kdb record stop`: Stops the current recording session. 
+* `kdb record delete [session_name]`: Deletes the specified recording session. If no session name is provided, the current session will be deleted. Deleting a session will not stop the recording. 
+* `kdb record list-sessions`: Lists all sessions by their name
+* `kdb record changes [session_name]`: Shows all changes for the specified session. If no session name is provided, the current session will be used. 
```

Comment:
Depending on what the actual format of `record export` looks like, it might be hard for a human to read. I think a simple listing of all changes like this could be quite helpful. We should, however, make it very clear that the output is not intended to be machine readable and might change at any time.

Reply:
Yes, as @kodebach mentioned the data displayed by this command is intended to be human-readable, as opposed to the machine-readable output of `record export`.

This feature is intended as a helpful reminder when debugging the currently changed configuration.

Reply:
I think this is `kdb record status` then? Changes, also in human-readable or statistical formats could be done by plugins, then it is also usable outside of `kdb record`.